presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -231,7 +231,7 @@ public Result apply(TableScanNode tableScanNode, Captures captures, Context cont
 
             return Result.ofPlanNode(new TableScanNode(
                     tableScanNode.getId(),
-                    tableScanNode.getTable(),
+                    tableScanNode.getTable().withLayout(Optional.of(layout.get().getLayout().getHandle().getConnectorHandle())),
```

Comment:
Curious about this `withLayout` mechanism. It looks to me in prestosql/presto#363, `getNewTableHandle` was introduced: https://github.com/prestosql/presto/pull/363/commits/47ce7e62491691eb91126c68db09f5eca1e080ff#diff-64a95c3a25c48aeecc5f744e3e45b7b8R236